### PR TITLE
Dedup receipt

### PR DESCRIPTION
**Pull request by arijitAD**

### CODE REVIEW

File: pkg/eth/cid_retriever.go [hunk 1]
```diff
@@ -316,12 +316,12 @@ func (ecr *CIDRetriever) RetrieveFilteredGQLLogs(tx *sqlx.Tx, rctFilter ReceiptF
...
 	pgStr := `SELECT eth.log_cids.leaf_cid, eth.log_cids.index, eth.log_cids.receipt_id,  
        			eth.log_cids.address, eth.log_cids.topic0, eth.log_cids.topic1, eth.log_cids.topic2, eth.log_cids.topic3, 
-       			eth.log_cids.log_data, eth.transaction_cids.tx_hash, data, eth.receipt_cids.cid, eth.receipt_cids.post_status
+       			eth.log_cids.log_data, eth.transaction_cids.tx_hash, data, eth.receipt_cids.leaf_cid as cid, eth.receipt_cids.post_status
 				FROM eth.log_cids, eth.receipt_cids, eth.transaction_cids, eth.header_cids, public.blocks
 				WHERE eth.log_cids.receipt_id = receipt_cids.id
 				AND receipt_cids.tx_id = transaction_cids.id
  				AND transaction_cids.header_id = header_cids.id
- 				AND log_cids.leaf_mh_key = blocks.key AND header_cids.block_hash = $1`
+ 				AND receipt_cids.leaf_mh_key = blocks.key AND header_cids.block_hash = $1`
```
Comment (i-norden) [inline pkg/eth/cid_retriever.go:324]: Brain is foggy and I'm not reasoning about this well atm, why are we changing this condition from `log_cids.leaf_mh_key = blocks.key` to  `receipt_cids.leaf_mh_key = blocks.key`?

File: pkg/eth/ipld_retriever.go [hunk 2]
```diff
@@ -366,23 +402,28 @@ func (r *IPLDRetriever) RetrieveReceiptsByBlockHash(hash common.Hash) ([]string,
 
 // RetrieveReceiptsByBlockNumber returns the cids and rlp bytes for the receipts corresponding to the provided block hash
```
Comment (i-norden) [inline pkg/eth/ipld_retriever.go:403]: We may want to note that the CIDs we are returning is not for the Receipts themselves but rather the leaf nodes they are contained within

File: pkg/eth/ipld_retriever.go [hunk 3]
```diff
@@ -366,23 +402,28 @@ func (r *IPLDRetriever) RetrieveReceiptsByBlockHash(hash common.Hash) ([]string,
...
+		if err != nil {
+			return nil, nil, err
+		}
+		rcts[i] = nodeVal
+		//rcts[i] = res.Data
 	}
 	return cids, rcts, nil
 }
 
 // RetrieveReceiptByHash returns the cid and rlp bytes for the receipt corresponding to the provided tx hash
```
Comment (i-norden) [inline pkg/eth/ipld_retriever.go:423]: We may want to note that the CID we are returning is not for the Receipt itself but rather the leaf node it was contained within

**Review by i-norden — APPROVED** [review]
Just a few minor comments and a question, thanks!

File: pkg/eth/filterer.go [hunk 4]
```diff
@@ -329,3 +322,44 @@ func checkNodeKeys(wantedKeys []common.Hash, actualKey []byte) bool {
 	}
 	return false
 }
+
+// FetchRctLeafNodeData fetches receipt leaf node IPLD data and CIDs
+func FetchRctLeafNodeData(rcts types.Receipts) ([]cid.Cid, [][]byte) {
```
[the remark this answers is not in the document]
Comment (arijitAD) [reply pkg/eth/filterer.go:327]: Done

File: pkg/eth/ipld_retriever.go [hunk 5]
```diff
@@ -356,8 +387,13 @@ func (r *IPLDRetriever) RetrieveReceiptsByBlockHash(hash common.Hash) ([]string,
...
 	for i, res := range rctResults {
-		cids[i] = res.CID
-		rcts[i] = res.Data
+		cids[i] = res.LeafCID
+		nodeVal, err := DecodeLeafNode(res.Data)
+		if err != nil {
+			return nil, nil, nil, err
+		}
+		rcts[i] = nodeVal
+		//rcts[i] = res.Data
```
[the remark this answers is not in the document]
Comment (arijitAD) [reply pkg/eth/ipld_retriever.go:396]: Done

File: pkg/eth/ipld_retriever.go [hunk 6]
```diff
@@ -366,23 +402,28 @@ func (r *IPLDRetriever) RetrieveReceiptsByBlockHash(hash common.Hash) ([]string,
...
 	for i, res := range rctResults {
-		cids[i] = res.CID
-		rcts[i] = res.Data
+		cids[i] = res.LeafCID
+		nodeVal, err := DecodeLeafNode(res.Data)
+		if err != nil {
+			return nil, nil, err
+		}
+		rcts[i] = nodeVal
+		//rcts[i] = res.Data
```
[the remark this answers is not in the document]
Comment (arijitAD) [reply pkg/eth/ipld_retriever.go:418]: Done

File: pkg/eth/test_helpers.go [hunk 7]
```diff
@@ -59,7 +59,7 @@ func TxModelsContainsCID(txs []models.TxModel, cid string) bool {
 // ListContainsBytes used to check if a list of byte arrays contains a particular byte array
```
[the remark this answers is not in the document]
Comment (arijitAD) [reply pkg/eth/test_helpers.go:59]: Done

File: pkg/eth/ipld_retriever.go (hunk 2 above)
Comment (arijitAD) [reply pkg/eth/ipld_retriever.go:403]: Done

File: pkg/eth/ipld_retriever.go (hunk 3 above)
Comment (arijitAD) [reply pkg/eth/ipld_retriever.go:423]: Done

File: pkg/eth/cid_retriever.go (hunk 1 above)
Comment (arijitAD) [reply pkg/eth/cid_retriever.go:324]: Reverted.

**Pull request merged**